### PR TITLE
Do not save un-persisted objects in workflow setup

### PR DESCRIPTION
**Pull request by cintamani**
https://eaflood.atlassian.net/browse/RUBY-352

The registration was saved at every refresh because of the workflow setup.
Since the object is persisted already at every `submit`, I have added a guard clause to avoid updating it unless it is persisted. 
BUT the editing journey is instead expecting a persisted object since the first call. Hence a fix there was necessary too.  

I wasn't able to find a nice way to test it :/ we should run some tests in truncation mode or do some weird stubbing and expectations.

### CODE REVIEW

File: app/controllers/waste_exemptions_engine/forms_controller.rb [hunk 1]
```diff
@@ -79,6 +79,8 @@ def set_workflow_state
       return unless state_can_navigate_flexibly?(@transient_registration.workflow_state)
       return unless state_can_navigate_flexibly?(requested_state)
```
Comment (cintamani) [inline app/controllers/waste_exemptions_engine/forms_controller.rb]:
```suggestion
```

File: app/forms/waste_exemptions_engine/edit_form.rb [hunk 2]
```diff
@@ -24,7 +24,10 @@ class EditForm < BaseForm
     # This form has a lot of attributes, so we have to disable the length cop.
     # rubocop:disable Metrics/MethodLength
     def initialize(registration)
+      registration.save!
```
Comment (cintamani) [inline app/forms/waste_exemptions_engine/edit_form.rb]:
```suggestion
      registration.save! unless registration.persisted?
```

File: app/forms/waste_exemptions_engine/edit_form.rb [hunk 3]
```diff
@@ -24,7 +24,10 @@ class EditForm < BaseForm
     # This form has a lot of attributes, so we have to disable the length cop.
     # rubocop:disable Metrics/MethodLength
```
Comment (cintamani) [inline app/forms/waste_exemptions_engine/edit_form.rb]:
```suggestion
    # rubocop:disable Metrics/MethodLength Metrics/AbcSize
```

File: app/controllers/waste_exemptions_engine/forms_controller.rb [hunk 4]
```diff
@@ -78,6 +78,7 @@ def setup_checks_pass?
     def set_workflow_state
       return unless state_can_navigate_flexibly?(@transient_registration.workflow_state)
       return unless state_can_navigate_flexibly?(requested_state)
+      return unless @transient_registration.persisted?
 
       @transient_registration.update_attributes(workflow_state: requested_state)
```
Comment (cintamani) [inline app/controllers/waste_exemptions_engine/forms_controller.rb]:
```suggestion
      @transient_registration.workflow_state = requested_state
```

File: app/controllers/waste_exemptions_engine/forms_controller.rb [hunk 5]
```diff
@@ -78,6 +78,7 @@ def setup_checks_pass?
     def set_workflow_state
       return unless state_can_navigate_flexibly?(@transient_registration.workflow_state)
       return unless state_can_navigate_flexibly?(requested_state)
+      return unless @transient_registration.persisted?
```
Comment (cintamani) [inline app/controllers/waste_exemptions_engine/forms_controller.rb]:
```suggestion
```

File: app/forms/waste_exemptions_engine/edit_form.rb [hunk 6]
```diff
@@ -23,8 +23,12 @@ class EditForm < BaseForm
 
     # This form has a lot of attributes, so we have to disable the length cop.
     # rubocop:disable Metrics/MethodLength
+    # rubocop:disable Metrics/AbcSize
     def initialize(registration)
+      registration.save! unless registration.persisted?
```
Comment (Cruikshanks) [inline app/forms/waste_exemptions_engine/edit_form.rb]: In todays (yesterdays 😪 ) meeting we said we were not going to touch the edit functionality. But can I assume this is still needed as it fixes something?

Comment (cintamani) [reply app/forms/waste_exemptions_engine/edit_form.rb]: it does fix something. We said we don't touch it but we need it persisted as it generates too many other fixes. At least this is what I recall 🤔

**Pull request merged**